sycl/source/detail/graph_impl.hpp

```diff
@@ -771,16 +771,28 @@ class partition {
   std::unordered_map<sycl::device, ur_exp_command_buffer_handle_t>
       MCommandBuffers;
   /// List of predecessors to this partition.
-  std::vector<std::shared_ptr<partition>> MPredecessors;
+  std::vector<std::weak_ptr<partition>> MPredecessors;
+
+  /// List of successors to this partition.
+  std::vector<std::weak_ptr<partition>> MSuccessors;
+
+  /// List of requirements for this partition.
+  std::vector<sycl::detail::AccessorImplHost *> MRequirements;
+
+  /// Storage for accessors which are used by this partition.
+  std::vector<AccessorImplPtr> MAccessors;
+
   /// True if the graph of this partition is a single path graph
   /// and in-order optmization can be applied on it.
   bool MIsInOrderGraph = false;
 
-  /// @return True if the partition contains a host task
-  bool isHostTask() const {
-    return (MRoots.size() && ((*MRoots.begin()).lock()->MCGType ==
-                              sycl::detail::CGType::CodeplayHostTask));
-  }
+  /// True if this partition contains only one node which is a host_task.
```

Comment:
```suggestion
  /// True if this partition contains at least one node which is a host_task.
```
hm or am I wrong and only one host task per partition is the limit? And if there is more than 1 host task, it belongs already to another partition?

Reply:
Makes sense